### PR TITLE
Featureimage/cmake migration

### PR DESCRIPTION
**Pull request by eric3873**

### CODE REVIEW

**Review by aquette — APPROVED** [review]
passes lintscript. Not done thorough analysis though!

File: resources/fty-metric-compute.conf.in [hunk 1]
```diff
@@ -1,3 +1,3 @@
 # create runtime directories for fty-metric-compute
-d /var/lib/fty/fty-metric-compute 0755 bios root
```
Comment (eric3873) [inline resources/fty-metric-compute.conf.in:2]: I think we can remove ExecStartPre section in the service definition

File: src/fty_mc_server.cc [hunk 2]
```diff
@@ -489,10 +489,10 @@ fty_mc_server_test (bool verbose)
 {
     printf (" * fty_mc_server:");
     printf ("\n");
-    assert(fty_shm_set_test_dir("src/selftest-rw") == 0);
+    assert(fty_shm_set_test_dir("selftest-rw") == 0);
```
Comment (jeffdegott) [inline src/fty_mc_server.cc:492]: prefer to use a #define or a const char*  for 'selftest_rw'

File: src/fty_metric_compute_private_selftest.cc [hunk 3]
```diff
@@ -42,10 +38,6 @@ fty_metric_compute_private_selftest (bool verbose, const char *subtest)
...
-}
-/*
-################################################################################
-#  THIS FILE IS 100% GENERATED BY ZPROJECT; DO NOT EDIT EXCEPT EXPERIMENTALLY  #
-#  Read the zproject/README.md for information about making permanent changes. #
-################################################################################
-*/
+    if (streq (subtest, "$ALL") || streq (subtest, "server_test"))
+        fty_mc_server_test (verbose);
+}
```
Comment (jeffdegott) [inline src/fty_metric_compute_private_selftest.cc:43]: ending \n

File: tests/main.cc [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+#include <catch2/catch.hpp>
+
+#include "../src/fty_metric_compute_classes.h"
+#include <filesystem>
+
+TEST_CASE("All the stuff of before")
+{
+    std::cout << "Current path is " << std::filesystem::current_path() << std::endl;
+    fty_metric_compute_private_selftest(true, "$ALL");
+}
```
Comment (jeffdegott) [inline tests/main.cc:26]: ending \n

**Review by jeffdegott — APPROVED** [review]

File: src/fty_mc_server.cc (hunk 2 above)
Comment (eric3873) [reply src/fty_mc_server.cc:492]: Take into account

File: src/fty_metric_compute_private_selftest.cc (hunk 3 above)
Comment (eric3873) [reply src/fty_metric_compute_private_selftest.cc:43]: Take into account

File: tests/main.cc (hunk 4 above)
Comment (eric3873) [reply tests/main.cc:26]: Take into account

**Pull request merged**